src/network/serverpackethandler.cpp

```diff
@@ -894,6 +898,24 @@ bool Server::checkInteractDistance(RemotePlayer *player, const f32 d, const std:
 	f32 max_d = BS * getToolRange(selected_item.getDefinition(m_itemdef),
 			hand_item.getDefinition(m_itemdef));
 
+	PlayerSAO *sao = player->getPlayerSAO();
+	if (sao) {
+		v3f attachment_pos;
+		int parent_id;
+		{
+			std::string bone;
+			v3f attachment_rot;
+			bool force_visible;
+			sao->getAttachment(&parent_id, &bone, &attachment_pos, &attachment_rot, &force_visible);
+		}
+		 if (parent_id) {
+			 // When attached, add the attachment radius to the allowed interaction range
+			 max_d += attachment_pos.getLength();
```

Comment:
so does the base position not include the correct attachment offset?

Reply:
The attachment offset consists of a transformation matrix which is currently not applied/calculated server-side. That would also require to synchronize auto-rotating items which currently just spin "randomly". I think it's good enough to simplify this as a spherical tolerance.

Reply:
I agree, was just wondering.